File: swift/config_curve.swift
/*
 * Copyright (c) 2012-2020 MIRACL UK Ltd.
 *
 * This file is part of MIRACL Core
 * (see https://github.com/miracl/core).
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

public struct CONFIG_CURVE{
    static public let WEIERSTRASS=0
    static public let EDWARDS=1
    static public let MONTGOMERY=2
    static public let NOT=0
    static public let BN=1
    static public let BLS12=2
    static public let BLS24=3
    static public let BLS48=4
    static public let D_TYPE=0
    static public let M_TYPE=1
    static public let POSITIVEX=0
    static public let NEGATIVEX=1

    static public let CURVETYPE = @CT@
    static public let CURVE_A = @CA@
    static public let CURVE_PAIRING_TYPE = @PF@
    static public let SEXTIC_TWIST = @ST@
    static public let SIGN_OF_X = @SX@
    static public let ATE_BITS = @AB@    
    static public let HTC_ISO = @HC@
    static public let HTC_ISO_G2 = @HC2@

    static public let HASH_TYPE = @HT@
    static public let AESKEY = @AK@

    static let ALLOW_ALT_COMPRESS = false

    static let USE_GLV = true
    static let USE_GS_G2 = true
    static let USE_GS_GT = true    
}

File: java/CONFIG_CURVE.java
/*
 * Copyright (c) 2012-2020 MIRACL UK Ltd.
 *
 * This file is part of MIRACL Core
 * (see https://github.com/miracl/core).
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.miracl.core.XXX;

public class CONFIG_CURVE {
    public static final int WEIERSTRASS = 0;
    public static final int EDWARDS = 1;
    public static final int MONTGOMERY = 2;
    public static final int NOT = 0;
    public static final int BN = 1;
    public static final int BLS12 = 2;
    public static final int BLS24 = 3;
    public static final int BLS48 = 4;
    public static final int D_TYPE = 0;
    public static final int M_TYPE = 1;
    public static final int POSITIVEX = 0;
    public static final int NEGATIVEX = 1;

    public static final int CURVETYPE = @CT@;
    public static final int CURVE_A = @CA@;
    public static final int CURVE_PAIRING_TYPE = @PF@;
    public static final int SEXTIC_TWIST = @ST@;
    public static final int SIGN_OF_X = @SX@;
    public static final int ATE_BITS = @AB@;
    public static final int G2_TABLE = @G2@;
    public static final int HTC_ISO = @HC@;
    public static final int HTC_ISO_G2 = @HC2@;

    public static final int SHA256 = 32;
    public static final int SHA384 = 48;
    public static final int SHA512 = 64;

    public static final int HASH_TYPE = @HT@;
    public static final int AESKEY = @AK@;

    public static final boolean ALLOW_ALT_COMPRESS = false;
    public static final boolean USE_GLV = true;
    public static final boolean USE_GS_G2 = true;
    public static final boolean USE_GS_GT = true;
}
